docs/design/nab_and_nar_status_update.md

```diff
@@ -0,0 +1,192 @@
+- `phase`
+- `conditions`
+- `veleroBackup` for NAB and `veleroRestore` for NAR, which contains name, namespace and status of the related Velero object.
+
+which are updated by NAB and NAR controllers.
+
+Only one update call should be performed per reconcile of NAB and NAR objects. Controller can requeue or depend on predicates if more updates are needed in that object.
+
+### Phase
+
+The `phase` field is a simple one high-level summary of the lifecycle of the objects, that only moves forward. Once a `phase` changes, it can not return to the previous value.
+
+It is always a one well defined value, that is intended to be a comprehensive state of a NAB or NAR object.
+
+Those are are the possible values for phase:
+
+| **Value** | **Description** |
+|-----------|-----------------|
+| New | *NonAdminBackup/NonAdminRestore* resource was accepted by the NAB/NAR Controller, but it has not yet been validated by the NAB/NAR Controller |
+| BackingOff | *NonAdminBackup/NonAdminRestore* resource was invalidated by the NAB/NAR Controller, due to invalid Spec. NAB/NAR Controller will not reconcile the object further, until user updates it |
+| Created | *NonAdminBackup/NonAdminRestore* resource was validated by the NAB/NAR Controller and Velero *Backup/restore* was created. The Phase will not have additional information about the *Backup/Restore* run |
+
+### Conditions
+
+The `conditions` field is a list of conditions.
+One NAB/NAR object may have multiple conditions.
+It is more granular knowledge of the NAB/NAR object and represents the array of the conditions through which the object has or has not passed.
+
+Each `condition` data is composed by the following:
+
+| **Field name** | **Description** |
+|----------------|-----------------|
+| type | The `NonAdminCondition` of the condition |
+| status | represents the state of individual condition. One of `True`, `False` or `Unknown`. |
+| lastTransitionTime | Timestamp for when this condition was created/updated. |
+| reason | Machine-readable, UpperCamelCase text indicating details about the last status transition. |
+| message | Human-readable message indicating details about the last status transition. |
+
+Those are are the possible values for `NonAdminCondition`:
+
+| **Value** | **Description** |
+|-----------|-----------------|
+| Accepted | The NonAdminBackup/NonAdminRestore object was accepted by the controller, but the Velero Backup/Restore may have not yet been created |
+| Queued | The Velero Backup/Restore was created successfully. At this stage errors may still occur either from the Velero not accepting object or during backup/restore procedure. |
+
+### Velero object reference
+
+NonAdminBackup/NonAdminRestore `status` contains reference to the related Velero Backup/Restore.
+
+NonAdminBackup `status.veleroBackup` contains `name`, `namespace` and `status`.
+- `status.veleroBackup.name` represents the name of the `VeleroBackup` object.
```

Comment:
in the NAB we in my opinion we should store our generated UUID of a Velero Object. The same UUID will be stored on the Velero side but as a Label, so the user won't be able to change the status on the NAB/NAR object easily.